src/web/rustdoc.rs

```diff
@@ -56,171 +49,205 @@ static DOC_RUST_LANG_ORG_REDIRECTS: Lazy<HashMap<&str, &str>> = Lazy::new(|| {
-pub fn rustdoc_redirector_handler(req: &mut Request) -> IronResult<Response> {
+#[instrument(skip_all)]
+pub(crate) async fn rustdoc_redirector_handler(
+    Path(params): Path<RustdocRedirectorParams>,
+    Extension(metrics): Extension<Arc<Metrics>>,
+    Extension(storage): Extension<Arc<Storage>>,
+    Extension(config): Extension<Arc<Config>>,
+    Extension(pool): Extension<Pool>,
+    Query(query_pairs): Query<HashMap<String, String>>,
+    uri: Uri,
+) -> AxumResult<impl IntoResponse> {
+    #[instrument]
     fn redirect_to_doc(
-        req: &Request,
+        query_pairs: &HashMap<String, String>,
         url_str: String,
         cache_policy: CachePolicy,
         path_in_crate: Option<&str>,
-    ) -> IronResult<Response> {
-        let mut queries = BTreeMap::new();
+    ) -> AxumResult<impl IntoResponse> {
+        let mut queries: BTreeMap<String, String> = BTreeMap::new();
         if let Some(path) = path_in_crate {
             queries.insert("search".into(), path.into());
         }
-        queries.extend(req.url.as_ref().query_pairs());
-        let url = ctry!(req, parse_url_with_params(&url_str, queries));
-        let mut resp = Response::with((status::Found, Redirect(url)));
-        resp.extensions.insert::<CachePolicy>(cache_policy);
-        Ok(resp)
-    }
-
-    fn redirect_to_crate(req: &Request, name: &str, vers: &str) -> IronResult<Response> {
-        let url = ctry!(
-            req,
-            Url::parse(&format!("{}/crate/{}/{}", redirect_base(req), name, vers)),
-        );
-
-        let mut resp = Response::with((status::Found, Redirect(url)));
-        resp.extensions
-            .insert::<CachePolicy>(CachePolicy::ForeverInCdn);
-        Ok(resp)
+        queries.extend(query_pairs.to_owned());
+        trace!("redirect to doc");
+        Ok(axum_cached_redirect(
+            axum_parse_uri_with_params(&url_str, queries)?,
+            cache_policy,
+        )?)
     }
 
-    let metrics = extension!(req, Metrics).clone();
     let mut rendering_time = RenderingTimesRecorder::new(&metrics.rustdoc_redirect_rendering_times);
 
-    // this unwrap is safe because iron urls are always able to use `path_segments`
-    // i'm using this instead of `req.url.path()` to avoid allocating the Vec, and also to avoid
-    // keeping the borrow alive into the return statement
-    if req
-        .url
-        .as_ref()
-        .path_segments()
-        .unwrap()
-        .last()
-        .map_or(false, |s| s.ends_with(".js"))
-    {
-        // javascript files should be handled by the file server instead of erroneously
-        // redirecting to the crate root page
-        if req.url.as_ref().path_segments().unwrap().count() > 2 {
-            // this URL is actually from a crate-internal path, serve it there instead
-            rendering_time.step("serve JS for crate");
-            return rustdoc_html_server_handler(req);
-        } else {
-            rendering_time.step("serve JS");
```

Comment:
I'm not sure which use-case this thing tries to handle. 

- release specific JS files were handled above 
- toolchain specific files were handled by the `LegacySharedResourcesHandler` 

So what remains are only files like `/krate/something.js` were we would look at exactly this path in the storage. 

I don't see where these could ever exist. 

The new code only handles to cases: 
1. crate-specific JS when we have name & version
2. toolchain static resources from the root

Reply:
I agree this looks like it's been obsoleted by refactors; I think it's ok to get rid of it.